### PR TITLE
Fix service environment issue

### PR DESCRIPTION
**Pull request by nsharma-git**
# Description

Ansible Runner APIs truncates larger string of service environment variable. As a result we have been seeing [unprintable] while trying to discover jolokia and jmx properties. This change now directly reads from the service and override files and creates environment variable map. 

Fixes # (issue)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Tested on molecule scenarios with Jolokia and Prometheus enabled. 

**Test Configuration**:

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] Any variable changes have been validated to be backwards compatible

### CODE REVIEW

File: discovery/manager/manager.py [hunk 1]
```diff
@@ -39,6 +39,8 @@ def get_ansible_vars_from_input_context(input_context: InputContext) -> dict:
         vars['ansible_ssh_extra_args'] = input_context.ansible_ssh_extra_args
         vars['ansible_python_interpreter'] = input_context.ansible_python_interpreter
         vars['ansible_ssh_private_key_file'] = input_context.ansible_ssh_private_key_file
+        vars['ansible_common_remote_group'] = input_context.ansible_common_remote_group
+        vars['ansible_become_password'] = input_context.ansible_become_password
```
Comment (mansisinha) [inline discovery/manager/manager.py:43]: are we also adding password flags? As of now, discovery expects password less ssh configuration via keys. Are we adding this enhancement?

Comment (nsharma-git) [reply discovery/manager/manager.py:43]: Yea, these are Ansible capabilities to connect to host. We are just proxies all possible parameters to Ansible runner.

File: discovery/manager/manager.py [hunk 2]
```diff
@@ -354,36 +374,6 @@ def get_log_file_path(input_context: InputContext, service: ServiceData, hosts:
 
         return None
 
-    @staticmethod
```
Comment (mansisinha) [inline discovery/manager/manager.py:357]: do we now have a new function equivalent to this one? How will we get the values of these env variables now?

Comment (nsharma-git) [reply discovery/manager/manager.py:357]: We have replaced this with AbstractPropertyBuilder.get_service_environment_variable which is more reliable and doesn't truncate the longer strings

File: discovery/service/service.py [hunk 3]
```diff
@@ -110,14 +113,48 @@ def get_service_facts(input_context: InputContext, service: ServiceData, hosts:
...
+        env_map = dict()
+
+        service_facts = AbstractPropertyBuilder.get_service_facts(input_context, service, hosts)
+        service_file = service_facts.get('FragmentPath')
+        override_file = service_facts.get('DropInPaths')
+
+        content = ServicePropertyManager.slurp_remote_file(input_context=input_context, hosts=hosts, file=service_file)
+        env_map.update(AbstractPropertyBuilder._get_systemd_env_details(content.get(hosts[0])))
+
+        content = ServicePropertyManager.slurp_remote_file(input_context=input_context, hosts=hosts, file=override_file)
```
Comment (mansisinha) [inline discovery/service/service.py:147]: is this called twice by design?

Comment (nsharma-git) [reply discovery/service/service.py:147]: Yes, we try to get the default service env variable and then override it by override file ( if present )

**Review by mansisinha — APPROVED** [review]
LGTM, Thanks for taking care of this

**Pull request merged**